util/fsdiffer/beforeafter_test.go

```diff
@@ -0,0 +1,113 @@
+package fsdiffer
+
+import (
+	"archive/tar"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// Do not test (at the moment) symlinks as every os needs a special syscall to set symlink times
+func TestBeforeAfterFSDiffer(t *testing.T) {
+	time1 := time.Now()
+	time2 := time.Now().Add(1 * time.Microsecond)
+	sourceFiles := []*buildFileInfo{
+		&buildFileInfo{path: "file01", typeflag: tar.TypeReg, mode: 0644, atime: time1, mtime: time1, contents: "hello"},
```

Comment:
How about some helpers to clean these up:
`func mkBuildFileInfoDir(path string, time time.Time) *buildFileInfo`
`func mkBuildFileInfoFile(path string, time time.Time, contents string) *buildFileInfo`